source/gitlab/gitlab.go

```diff
@@ -113,13 +113,30 @@ func WithInstance(client *gitlab.Client, config *Config) (source.Driver, error)
 }
 
 func (g *Gitlab) readDirectory() error {
-	nodes, response, err := g.client.Repositories.ListTree(g.projectID, g.listOptions)
-	if err != nil {
-		return err
-	}
+	var (
+		nodes    []*gitlab.TreeNode
+		n        []*gitlab.TreeNode
+		response *gitlab.Response
+		err      error
+	)
+
+	for {
+		n, response, err = g.client.Repositories.ListTree(g.projectID, g.listOptions)
+		if err != nil {
+			return err
+		}
+
+		if response.StatusCode != http.StatusOK {
```

Comment:
Note, this will fail the whole `readDirectory()` call if a single request in the request chain fails.
To reduce the chances of that happening, it may be worth defaulting the `per_page` option to `100` (the [max](https://gitlab.com/help/api/README.md#offset-based-pagination)) and making that configurable.

Reply:
How would you recommend to configure that parameter?

Reply:
Use a [query parameter](https://golang.org/pkg/net/url/#URL.Query)

Reply:
Oh, I mean how to pass that parameter from `migrate.go` to a particular gitlab driver. Do you think extending `Gitlab.url` string is a good idea? For instance, `gitlab://username:token@gitlab.com/11197284/migrations?per_page=50`

Reply:
If you're using `migrate` via the CLI, the only way to configure the source driver is via URL query params. We try our best to avoid conflicts with the underlying driver, but for gitlab, that doesn't seem like it'll be an issue.

If you're using `migrate` as a Go library, the driver can be configured via `WithInstance()`. It looks like gitlab isn't properly using `Config`